doc/adr/0004-use-cognito-for-authentication.md

```diff
@@ -0,0 +1,47 @@
+# 4. Use Cognito for authentication (via APIs)
+
+Date: 2019-08-12
+
+## Status
+
+Accepted
+
+## Context
+
+Given our application allows making changes to the federation configuration,
+it is neccessary to secure the service appropriately. The same instance
+will be accessed by different users with varying level of permissions.
+Therefore we need to have a strong authentication and user management in place.
+Our key authentication requirements are:
+- Secure
+- Flexible and extendable
+- Cost effective
+- Multi-factor authentication (MFA) support
+- GDPR compliant
+
+## Decision
+
+Based on our requirements, we decided to use Amazon Cognito - a service provided by AWS.
+Amazon Cognito satisfies all the requirements and offers an easy way to implement 
```

Comment:
'offers an easy way to implement authentication in our application' not sure about this one as it wasn't the easiest compared to say, regular devise and a users table

Reply:
I think i agree, it would have been better if it offered fully ui customisation. So we have had to work around this

Reply:
> 'offers an easy way to implement authentication in our application' not sure about this one as it wasn't the easiest compared to say, regular devise and a users table

I think I meant that if we went for the OAuth flow. The initial implementation was really easy (with OmniAuth) when we used OAuth. Once we changed to API, it's true it was harder to set up. But now any extension, we just call the SDK and it's easy again.